src/main/java/com/aws/greengrass/deployment/converter/DeploymentDocumentConverter.java

```diff
@@ -139,54 +147,57 @@ private static List<DeploymentPackageConfiguration> buildDeploymentPackageConfig
-                        .timestamp(config.getCreationTimestamp());
+        DeploymentDocument.DeploymentDocumentBuilder builder = DeploymentDocument.builder()
+                .configurationArn(config.getConfigurationArn()).deploymentId(config.getDeploymentId())
+                .requiredCapabilities(config.getRequiredCapabilities())
+                .deploymentPackageConfigurationList(convertComponents(config.getComponents()))
+                .groupName(parseGroupNameFromConfigurationArn(config)).timestamp(config.getCreationTimestamp());
 
         convertThingGroupArns(builder, config);
 
         if (config.getFailureHandlingPolicy() == null) {
-            // FailureHandlingPolicy should be provided per contract with CreateDeployment API.
+            // FailureHandlingPolicy should be provided per contract with CreateDeployment
+            // API.
             // However if it is not, device could proceed with default for resilience.
             logger.atWarn().log("FailureHandlingPolicy should be provided but is not provided. "
-                                        + "Proceeding with default failure handling policy.");
+                    + "Proceeding with default failure handling policy.");
         } else {
             builder.failureHandlingPolicy(convertFailureHandlingPolicy(config.getFailureHandlingPolicy()));
         }
 
         if (config.getComponentUpdatePolicy() == null) {
-            // ComponentUpdatePolicy should be provided per contract with CreateDeployment API.
+            // ComponentUpdatePolicy should be provided per contract with CreateDeployment
+            // API.
             // However if it is not, device could proceed with default for resilience.
             logger.atWarn().log("ComponentUpdatePolicy should be provided but is not provided. "
-                                        + "Proceeding with default failure handling policy.");
+                    + "Proceeding with default failure handling policy.");
         } else {
             builder.componentUpdatePolicy(convertComponentUpdatePolicy(config.getComponentUpdatePolicy()));
         }
 
         if (config.getConfigurationValidationPolicy() == null) {
-            // ConfigurationValidationPolicy should be provided per contract with CreateDeployment API.
+            // ConfigurationValidationPolicy should be provided per contract with
+            // CreateDeployment API.
             // However if it is not, device could proceed with default for resilience.
             logger.atWarn().log("ConfigurationValidationPolicy should be provided but is not provided. "
                     + "Proceeding with default failure handling policy.");
         } else {
-            builder.configurationValidationPolicy(convertConfigurationValidationPolicy(
-                    config.getConfigurationValidationPolicy())
-            );
+            builder.configurationValidationPolicy(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Potential log injection detected. Ensure all untrusted input is properly sanitized before logging. Use parameterized logging or validate input against a allow list to prevent log injection vulnerabilities. Consider using a dedicated logging library&#39;s built-in sanitization features when available. Learn more - https://cwe.mitre.org/data/definitions/117.html